storybook/src/components/Text.tsx

```diff
@@ -0,0 +1,21 @@
+import React from 'react'
+
+// 요런 식으로 작성하면 되나 ..?!
+interface TextProps {
+  children: JSX.Element
+  size: number
+  strong: boolean
+  [other: string]: any
+}
+
+// 이 React.FC는 뭐지 ..? 매개변수에 타입 적용해주면 안되나? ㅠㅠ
+const Text: React.FC<TextProps> = ({ children, size, strong, ...props }) => {
+  const fontStyle = {
+    fontWeight: strong ? 'bold' : undefined,
+    fontSize: size,
+  }
+
+  return <div style={{ ...props.style, ...fontStyle }}>{children}</div>
+}
```

Comment:
저도 아무 생각없이 `React.FC`를 접하고 있어서 한 번 GPT에게 물어봤습니다..! 대략적으로 다음과 같이 이해하면 좋을 것 같네요!!

> React.FC는 React의 TypeScript 타입에서 함수 컴포넌트를 나타내는 타입 별칭입니다.
> React.FC 컴포넌트는 props를 인수로 받아 ReactNode를 반환해야 합니다. ReactNode는 React에서 렌더링할 수 있는 모든 값을 나타냅니다. 예를 들어 문자열, 숫자, React 요소 또는 함수입니다.

그리고 제 경험적으로 이야기하면 주로 `interface`가 아닌 `type`을 쓸 때 저렇게 많이 쓰더라구요!!

```suggestion
interface TextProps {
  children: JSX.Element
  size: number
  strong: boolean
  [other: string]: any
}
const Text = ({ children, size, strong, ...props }: TextProps) => {
  const fontStyle = {
    fontWeight: strong ? 'bold' : undefined,
    fontSize: size,
  }

  return <div style={{ ...props.style, ...fontStyle }}>{children}</div>
}
```

이렇게 사용해도 동일한 것으로 이해하고 있습니다..! 🙇‍♂️
이와 관련해서 `interface`를 쓸지 `type`을 쓸지는 팀 내에 이야기하는 부분인 것 같네요!

Reply:
네네 React.FC는 단점이 많아서 인자의 타입을 일일이 정해주는게 좋다는 글이 많더라구요!